sdk/identity/identity/src/msal/nodeFlows/msalOpenBrowser.ts

```diff
@@ -36,218 +30,39 @@ export const interactiveBrowserMockable = {
-          reject(
-            new CredentialUnavailableError(
-              [
-                `InteractiveBrowserCredential: Access denied to port ${this.port}.`,
-                `Try sending a redirect URI with a different port, as follows:`,
-                '`new InteractiveBrowserCredential({ redirectUri: "http://localhost:1337" })`',
-              ].join(" ")
-            )
-          );
-        } else {
-          reject(
-            new CredentialUnavailableError(
-              `InteractiveBrowserCredential: Failed to start the necessary web server. Error: ${err.message}`
-            )
-          );
-        }
-      });
-
-      app.on("listening", () => {
-        const openPromise = this.openAuthCodeUrl(scopes, options);
-
-        const abortSignal = options?.abortSignal;
-        if (abortSignal) {
-          abortSignal.addEventListener("abort", () => {
-            cleanup();
-            reject(new Error("Aborted"));
-          });
-        }
-
-        openPromise.catch((e) => {
-          cleanup();
-          reject(e);
-        });
-      });
-    });
-  }
-
-  private pkceCodes?: {
-    verifier: string;
-    challenge: string;
-  };
-
-  private async openAuthCodeUrl(
-    scopeArray: string[],
-    options?: CredentialFlowGetTokenOptions
-  ): Promise<void> {
-    // Initialize CryptoProvider instance
-    const cryptoProvider = new msalNode.CryptoProvider();
-    // Generate PKCE Codes before starting the authorization flow
-    this.pkceCodes = await cryptoProvider.generatePkceCodes();
-
-    const authCodeUrlParameters: msalNode.AuthorizationUrlRequest = {
-      scopes: scopeArray,
-      correlationId: options?.correlationId,
-      redirectUri: this.redirectUri,
+    const response = await this.getApp("public", options?.enableCae).acquireTokenInteractive({
+      openBrowser: async (url) => {
+        await interactiveBrowserMockable.open(url, { wait: true, newInstance: true });
+      },
+      scopes,
       authority: options?.authority,
       claims: options?.claims,
+      correlationId: options?.correlationId,
       loginHint: this.loginHint,
-      codeChallenge: this.pkceCodes.challenge,
-      codeChallengeMethod: "S256", // Use SHA256 Algorithm
-    };
-    const response = await this.getApp("public", options?.enableCae).getAuthCodeUrl(
-      authCodeUrlParameters
-    );
-    try {
-      // A new instance on macOS only which allows it to not hang, does not fix the issue on linux
-      await interactiveBrowserMockable.open(response, { wait: true, newInstance: true });
-    } catch (e: any) {
-      throw new CredentialUnavailableError(
-        `InteractiveBrowserCredential: Could not open a browser window. Error: ${e.message}`
+    });
+
+    const expiresOnTimestamp = response?.expiresOn?.valueOf();
+    if (!expiresOnTimestamp) {
+      throw new Error(
+        `Interactive Browser Authentication Error "Did not receive token with a valid expiration"`
       );
     }
+
+    return {
```

Comment:
You can make the overally structure for result and errors similar to the `doGetToken` of this file - `src\msal\nodeFlows\msalAuthorizationCode.ts`